iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/CreateManyDatabasesProcedure.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.confignode.procedure.impl;
+
+import org.apache.iotdb.common.rpc.thrift.TSStatus;
+import org.apache.iotdb.commons.utils.TestOnly;
+import org.apache.iotdb.confignode.consensus.request.ConfigPhysicalPlanType;
+import org.apache.iotdb.confignode.consensus.request.write.database.DatabaseSchemaPlan;
+import org.apache.iotdb.confignode.manager.ProcedureManager;
+import org.apache.iotdb.confignode.procedure.env.ConfigNodeProcedureEnv;
+import org.apache.iotdb.confignode.procedure.exception.ProcedureException;
+import org.apache.iotdb.confignode.procedure.impl.statemachine.StateMachineProcedure;
+import org.apache.iotdb.confignode.procedure.store.ProcedureType;
+import org.apache.iotdb.confignode.rpc.thrift.TDatabaseSchema;
+import org.apache.iotdb.rpc.TSStatusCode;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+/**
+ * This procedure will create numerous databases (perhaps 100), during which the confignode leader
+ * should be externally shutdown to test whether the procedure can be correctly recovered after the
+ * leader change. If the procedure is still not recovered by the time the last database is created,
+ * it will throw an exception which indicates the test to be poorly written.
+ */
+@TestOnly
+public class CreateManyDatabasesProcedure
+    extends StateMachineProcedure<ConfigNodeProcedureEnv, Integer> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(CreateManyDatabasesProcedure.class);
+  public static final int MAX_STATE = 100;
+  public static final String DATABASE_NAME_PREFIX = "root.test_";
+  public static final int SLEEP_INTERVAL = 1000;
+  private boolean createFailedOnce = false;
+
+  @Override
+  protected Flow executeFromState(ConfigNodeProcedureEnv configNodeProcedureEnv, Integer state)
+      throws InterruptedException {
+    if (state < MAX_STATE) {
+      if (state == MAX_STATE - 1 && !isDeserialized()) {
```

Comment:
"Test code" may not ensure every recover of the procedure, since the running environment varies and the threads are executed in random conditions. Besides, whether the test code is valid enough matters only to tests and may not to the running cluster. Still I think it's better to remove this judgement.